syncano/models/fields.py

```diff
@@ -0,0 +1,270 @@
+import re
+import six
+from datetime import date, datetime
+
+from syncano.exceptions import SyncanoFieldError
+from .manager import RelatedManagerDescriptor
+
+
+class Field(object):
+    required = False
+    read_only = True
+    default = None
+
+    def __init__(self, name=None, **kwargs):
+        self.name = name
+        self.model = None
+        self.label = kwargs.pop('label', None)
+        self.default = kwargs.pop('default', None)
+        self.max_length = kwargs.pop('max_length', None)
+        self.min_length = kwargs.pop('min_length', None)
+
+        self.schema = kwargs
+
+    def __get__(self, instance, owner):
+        return instance._raw_data.get(self.name, self.default)
+
+    def __set__(self, instance, value):
+        self.validate(value, instance)
+        instance._raw_data[self.name] = self.to_python(value)
+
+    def __delete__(self, instance):
+        if self.name in instance._raw_data:
+            del instance._raw_data[self.name]
+
+    def validate(self, value, model_instance):
+        if self.required and not value:
+            raise self.VaidationError('This field is required.')
+
+        if self.read_only and getattr(model_instance, self.name):
+            raise self.VaidationError('Field is read only.')
+
+        if isinstance(value, six.string_types):
+            if self.max_length and len(value) > self.max_length:
+                raise self.VaidationError('Max length reached')
+
+            if self.min_length and len(value) < self.min_length:
+                raise self.VaidationError('Max length reached')
+
+    def to_python(self, value):
+        return value
+
+    def to_native(self, value):
+        return value
+
+    def contribute_to_class(self, cls, name):
+        self.model = cls
+        cls._meta.add_field(self)
+
+        if not self.name:
+            self.name = name
+
+        setattr(cls, name, self)
+
+        ErrorClass = type(
+            '{0}VaidationError'.format(self.__class__.__name__),
+            (SyncanoFieldError, ),
+            {'field_name': name}
+        )
+
+        setattr(self, 'VaidationError', ErrorClass)
+
+
+class WritableField(Field):
+    required = True
+    read_only = False
+
+    def __init__(self, *args, **kwargs):
+        super(WritableField, self).__init__(*args, **kwargs)
+        self.required = kwargs.pop('required', True)
+        self.read_only = kwargs.pop('read_only', False)
+
+
+class StringField(WritableField):
+
+    def to_python(self, value):
+        if isinstance(value, six.string_types) or value is None:
+            return value
+        return six.u(value)
+
+
+class IntegerField(WritableField):
+
+    def to_python(self, value):
+        if value is None:
+            return value
+        try:
+            return int(value)
+        except (TypeError, ValueError):
+            raise self.VaidationError('Invalid value.')
```

Comment:
Maybe information about expected type would be helpful? For example: `'Invalid value. Value should be integer.'`